### ISSUE TITLE
Dev mode persists checks (docker)

### ISSUE THREAD

**Issue by eaglo** [hashicorp/consul#4377]
#### Overview of the Issue

We are using dev mode with a single consul node for our local development stack. With the new release we see that even after restarting the consul docker instance the checks persist while it says in the documentation that dev mode does not persist anything.

#### Reproduction Steps

Steps to reproduce this issue, eg:

1. Run a consul docker instance
1. Register a check
1. Restart docker instance, see that the check is still there

### Operating system and Environment details

Windows 10, Docker Instance

**Comment by eaglo** [comment]
Yup, the docker image is in fact adding a volume. But I think it is like that for almost three years :)

https://github.com/hashicorp/docker-consul/blob/75d540650a025b682daa8f0760cdce02c5318602/0.X/Dockerfile#L50

In this case i would suggest to fix the documentation...?

**Comment by mitchellh** [comment]
No I think the Docker image is doing the right thing, it hasn't changed, the only thing that changed and is causing this is noted in my comment above. But as I said I'm not sure on the right way forward at the moment but will discuss with the Consul team. :)
